docpages/building/linux.md

```diff
@@ -18,12 +19,14 @@ cd build
 sudo make install
 ```
 
+\note If you are on Arch Linux, follow the steps below and set the path to `/usr/` to avoid shared-library conflict.
```

Comment:
is this actually correct advice?
what happens if the user already has it installed from pacman or aur? it'd overwrite it and may leave the system in a corrupt state.

Reply:
if user already has installed from pacman
 1. they wont follow this step
 2. if they follow because they are dum dum it will clearly show that `/usr/include/dpp is already located no changes has been made`

Reply:
Let me send the log on it when i install it from aur and then try to build it from source

Reply:
> if user already has installed from pacman
> 
>     1. they wont follow this step
> 
>     2. if they follow because they are dum dum it will clearly show that `/usr/include/dpp is already located no changes has been made`

but what if they DID?
/usr is never from hand built programs, ever, in any linux distribution.

Reply:
i installed it from aur, then i build it from source so it gave output
```
-- Up-to-date: /usr/include/dpp/coro
-- Installing: /usr/include/dpp/coro/job.h
```
it means it updated some outdated package and then it created whats new, if we build it from source again it will only show `up-to-date`

Reply:
> > if user already has installed from pacman
> > ```
> > 1. they wont follow this step
> > 
> > 2. if they follow because they are dum dum it will clearly show that `/usr/include/dpp is already located no changes has been made`
> > ```
> 
> but what if they DID? /usr is never from hand built programs, ever, in any linux distribution.

if u wont set /usr/ on arch it wont work, g++ in standard arch will look for `/usr/include` its arch linux issue and this can be confirmed by an old issue https://discord.com/channels/825407338755653642/825411707521728512/964127822216069150

Reply:
Please see the FHS:
https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s05.html#specificOptions12

Reply:
> > > if user already has installed from pacman
> > > ```
> > > 1. they wont follow this step
> > > 
> > > 2. if they follow because they are dum dum it will clearly show that `/usr/include/dpp is already located no changes has been made`
> > > ```
> > 
> > 
> > but what if they DID? /usr is never from hand built programs, ever, in any linux distribution.
> 
> if u wont set /usr/ on arch it wont work, g++ in standard arch will look for `/usr/include` its arch linux issue and this can be confirmed by an old issue https://discord.com/channels/825407338755653642/825411707521728512/964127822216069150

it's a hack and a bad workaround, the only correct way for self built lib is use it directly by providing appropriate compile flags. In cmake by using `target_link_directories` command, or directly in the compiler flag `-L/path/to/lib/dir/`